.github/workflows/checks.yml

```
@@ -99,13 +99,37 @@ jobs:
         with:
           app-id: ${{ secrets.WORKFLOW_STOPPER_RUNNER_APP_ID }}
           app-key: ${{ secrets.WORKFLOW_STOPPER_RUNNER_APP_KEY }}
-  # name of this job must be unique across all workflows
-  # otherwise GitHub will mark all these jobs as required
+
+  cargo-udeps:
+    runs-on: ${{ needs.preflight.outputs.RUNNER }}
+    needs: [preflight]
+    if: ${{ needs.preflight.outputs.changes_rust }}
+    timeout-minutes: 60
+    container:
+      image: ${{ needs.preflight.outputs.IMAGE }}
+    env:
+      SKIP_WASM_BUILD: 1
+    steps:
+      - uses: actions/checkout@6d193bf28034eafb982f37bd894289fe649468fc # v4.1.7
+      - name: Run cargo-udeps (workspace)
+        id: required
+        run: |
+          if command -v forklift >/dev/null 2>&1; then
+            forklift cargo +nightly udeps --workspace --all-features --all-targets --exclude polkadot-zombienet-sdk-tests
+          else
+            cargo +nightly udeps --workspace --all-features --all-targets --exclude polkadot-zombienet-sdk-tests
+          fi
```

```suggestion
      - name: Run cargo-udeps (workspace)
        id: required
        run: |
          cargo install cargo-udeps --locked
          cargo +nightly udeps --workspace --all-features --all-targets --exclude polkadot-zombienet-sdk-tests
```
There is no need to use forklift with `cargo-udeps`